### PR TITLE
Tag unit test refactor from Enzyme to RTL #1797

### PR DESCRIPTION
**Pull request by Tera15**
# Description

Refactored unit tests for the Tag component.

--

[1797](https://github.com/ifmeorg/ifme/issues)

---

Reviewing this pull request? Check out our [Code Review Practices](https://github.com/ifmeorg/ifme/wiki/Code-Review-Practices) guide if you haven't already!

### CODE REVIEW

File: client/app/components/Resource/__tests__/Resource.spec.jsx [hunk 1]
```diff
@@ -17,68 +17,49 @@ const AUTHOR = 'Desi Rottman';
...
-    expect(wrapper.find('.author').exists()).toEqual(false);
-    expect(wrapper.find('.tags').exists()).toEqual(false);
-
-    wrapper = shallow(
-      <Resource title={TITLE} link={URL} tags={TAGS} author={AUTHOR} />,
-    );
-    expect(wrapper.find('.author').exists()).toEqual(false);
-    expect(wrapper.find('.tags').exists()).toEqual(false);
+  const { getByRole, getByText } = screen;
+  test('it renders with neither tags nor author', () => {
```
Comment (julianguyen) [inline client/app/components/Resource/__tests__/Resource.spec.jsx:21]: I would rephrase this as: it renders with a title and link.

File: client/app/components/Resource/__tests__/Resource.spec.jsx [hunk 2]
```diff
@@ -17,68 +17,49 @@ const AUTHOR = 'Desi Rottman';
...
-    expect(wrapper.find('Tag').length).toEqual(TAGS.length);
-
-    wrapper = shallow(
-      <Resource title={TITLE} link={URL} tagged tags={EMPTY_TAGS} />,
-    );
-    expect(wrapper.find('.author').exists()).toEqual(false);
-    expect(wrapper.find('.tags').exists()).toEqual(true);
-    expect(wrapper.find('Tag').length).toEqual(0);
+  test('it renders with tags={TAGS} and no tagged prop', () => {
+    render(<Resource title={TITLE} link={URL} tags={TAGS} />);
```
Comment (julianguyen) [inline client/app/components/Resource/__tests__/Resource.spec.jsx:27]: Taking a look at the Resources component it seems like tagged is a redundant prop. If `tags` is undefined then that should be enough to tell us that there are no tags. We don't need `tagged`.

I would refactor the [component](https://github.com/ifmeorg/ifme/blob/master/client/app/components/Resource/index.jsx#L12) to simplify this logic and then update this test.

File: client/app/components/Resource/__tests__/Resource.spec.jsx [hunk 3]
```diff
@@ -17,68 +17,49 @@ const AUTHOR = 'Desi Rottman';
...
-    expect(wrapper.find('.tags').exists()).toEqual(true);
-    expect(wrapper.find('Tag').length).toEqual(0);
-
-    wrapper = shallow(
+  test('it renders with tags={EMPTY_TAGS} and no author ', () => {
+    render(<Resource title={TITLE} link={URL} tagged tags={EMPTY_TAGS} />);
+    expect(getByRole('link')).toBeInTheDocument();
+    expect(getByText(TITLE)).toBeInTheDocument();
+  });
+  test('it renders with external and author={AUTHOR}', () => {
```
Comment (julianguyen) [inline client/app/components/Resource/__tests__/Resource.spec.jsx:42]: In a similar tone as above, we don't need `external` and the `author` prop should be enough.

File: client/app/components/Tag/__tests__/Tag.spec.jsx [hunk 4]
```diff
@@ -1,28 +1,41 @@
...
-      );
-    });
-
-    it('supports secondary class', () => {
-      const tag = shallow(<Tag secondary />);
-      expect(tag.prop('className')).toEqual(
-        expect.stringContaining(css.secondary),
-      );
-    });
+describe('Tooltip', () => {
```
Comment (julianguyen) [inline client/app/components/Tag/__tests__/Tag.spec.jsx:9]: These tests are looking good!

**Comment by Tera15** [comment]
Cool, that actually makes a lot of sense to refactor the Resources component. I will take a deeper look at everything tomorrow and let you know if I have any questions. Thanks!

File: client/app/components/Resource/__tests__/Resource.spec.jsx (hunk 2 above)
Comment (Tera15) [reply client/app/components/Resource/__tests__/Resource.spec.jsx:27]: I'm going to get started on all of this tonight :) I'll reach out and let you know if I have any questions !

**Comment by Tera15** [comment]
Taking your advice I removed the tagged and external props resulting in quite a large reduction in lines of code in the test file. Let me know what you think / any other changes you'd like :)

**Comment by julianguyen** [comment]
@Tera15 It looks like there's some merge conflicts! Could you resolve them and then I'll re-review!

**Comment by Tera15** [comment]
Oh I'm sorry I totally missed that. I sure will :)

**Comment by Tera15** [comment]
@julianguyen  Conflicts are resolved!

**Review by julianguyen — APPROVED** [review]
Excellent work! Thanks so much as usual :D
